### ISSUE TITLE
Profile the engine binary

### ISSUE THREAD

**Issue by mominul** [OpenBangla/OpenBangla-Keyboard#60]
Profile the engine binary to detect the code hotspots.

**Comment by Adyel** [comment]
@mominul  How are you going to profile it? I just check and It needs significant knowledge and understanding about C++.

Here is My report using [Intel® VTune™ Amplifier](https://software.intel.com/en-us/intel-vtune-amplifier-xe/). 

![image](https://user-images.githubusercontent.com/4653623/43257835-abdfdcfc-90f2-11e8-9b76-cffe2fc262b7.png)

![image](https://user-images.githubusercontent.com/4653623/43257867-bc5cb474-90f2-11e8-88ef-3d492d5da310.png)

**Comment by mominul** [comment]
Thanks for the work @Adyel ! I wanted to use the `perf` binary comes with linux. But yours profiler is very powerful indeed! Are you using a trial version?

The report is somewhat I was expecting. When I wrote the whole [layout handling code](https://github.com/OpenBangla/OpenBangla-Keyboard/tree/master/src/engine/libengine), I was following 'if it compiles and works, let's have it' method. :laughing: So we need to rewrite the whole stuff with the meaningful methods and with performance kept in mind. I have already started the work [here](https://github.com/OpenBangla/libRiti). I am thinking to implement #61 in libRiti. I want to make libRiti somewhat like [libchewing](https://github.com/chewing/libchewing).
